examples/nlp/language_modeling/conf/megatron_gpt_config.yaml

```diff
@@ -24,6 +24,16 @@ trainer:
   benchmark: False
   enable_model_summary: False # default PTL callback for this does not support model parallelism, instead we log manually
 
+# Used for S3 Checkpointing
+checkpointing:
+  # write_concurrency * tp * pp * 1.15 (buffer) should be within 3500 S3 TPS limit per partition
+  max_write_concurrency: 10
+  # read_concurrency * tp * pp * 1.15 (buffer) should be within 5500 S3 TPS limit per partition
+  max_read_concurrency: 15
+  chunk_size_MB: 64
+  # enables asynchronous checkpoint writing to S3 dirpath
+  enable_async_checkpointing: False
```

Comment:
General comment to async checkpointing: we can't just start a background save and forget, because we lose consistency guarantees (e.g. old `-last` checkpoint will be removed immediately, but we don't know if the ongoing ckpt save will succeed).

I implemented this mechanism, it's in review now. It's only for distributed checkpoints, but I think it could be relatively easily adjusted to this approach.

Reply:
It is not included in this PR but our workload implements a `save_last_k` flag which enables us to save, for example, the latest 3 checkpoints. 

The checkpointing in this PR is for the legacy format; could you link a reference to your mechanism you implemented, and your thoughts on whether it is necessary for the legacy format (I know that there are -unfinished checkpoints in the distributed checkpoint, but am not sure if it's related to async saving)?

Reply:
> could you link a reference to your mechanism you implemented,

You can track the usage of this flag: https://github.com/NVIDIA/NeMo/blob/main/nemo/utils/callbacks/nemo_model_checkpoint.py#L55

Main part is here: https://github.com/NVIDIA/NeMo/blob/main/nemo/utils/callbacks/nemo_model_checkpoint.py#L455


My main concern is that until an async ckpt is completed, you shouldn't remove the old ones, but I guess saving top 3 ckpts helps with that. But in that case you need to ensure or at least document that fact.

Reply:
Ok got it. 
I have default usage set to false for async, and I can doc in both the class descriptor, rst, and config yaml of this concern.

Reply:
thanks